### PR TITLE
Modification to evm-semantics for single opcode proof

### PR DESCRIPTION
**Pull request by yzhang90**

### CODE REVIEW

File: data.md [hunk 1]
```diff
@@ -430,14 +431,14 @@ A cons-list is used for the EVM wordstack.
 -   `WS [ N := W ]` sets element $N$ of $WS$ to $W$ (padding with zeros as needed).
 
 ```k
-    syntax Int ::= WordStack "[" Int "]" [function]
- // -----------------------------------------------
+    syntax Int ::= WordStack "[" Int "]" [function, functional]
```
Comment (ttuegel) [inline data.md]: I don't think this is `functional`, is it? What if `N <Int 0`?

File: data.md [hunk 2]
```diff
@@ -430,14 +431,14 @@ A cons-list is used for the EVM wordstack.
...
+ // -----------------------------------------------------------
     rule (W0 : WS)   [N] => W0           requires N ==Int 0
     rule (.WordStack)[N] => 0            requires N >Int 0
     rule (W0 : WS)   [N] => WS[N -Int 1] requires N >Int 0
 
-    syntax WordStack ::= WordStack "[" Int ":=" Int "]" [function]
- // --------------------------------------------------------------
+    syntax WordStack ::= WordStack "[" Int ":=" Int "]" [function, functional]
```
Comment (ttuegel) [inline data.md]: As above, is this really `functional`?

File: evm.md [hunk 3]
```diff
@@ -1870,9 +1870,9 @@ In the YellowPaper, each opcode is defined to consume zero gas unless specified
                      orBool OP ==K RETURN
                      orBool OP ==K REVERT
 
-    syntax Int ::= #memoryUsageUpdate ( Int , Int , Int ) [function]
- // ----------------------------------------------------------------
-    rule #memoryUsageUpdate(MU, START, WIDTH) => MU                                       requires WIDTH ==Int 0
+    syntax Int ::= #memoryUsageUpdate ( Int , Int , Int ) [function, functional]
+ // ----------------------------------------------------------------------------
+    rule #memoryUsageUpdate(MU, START, WIDTH) => MU                                       requires WIDTH <=Int 0
```
Comment (ehildenb) [inline evm.md]: `notBool WIDTH >Int 0`

Comment (ehildenb) [inline evm.md]:
```suggestion
    rule #memoryUsageUpdate(MU, START, WIDTH) => MU                                       requires notBool WIDTH >Int 0
```

File: evm.md [hunk 4]
```diff
@@ -1014,8 +1014,8 @@ Otherwise, it is calculated here using the "shortcut" formula used for running t
          <number>      HI     </number>
          <blockhashes> HASHES </blockhashes>
 
-    syntax Int ::= #blockhash ( List , Int , Int , Int ) [function]
- // ---------------------------------------------------------------
+    syntax Int ::= #blockhash ( List , Int , Int , Int ) [function, functional]
+ // ---------------------------------------------------------------------------
```
Comment (ehildenb) [inline evm.md]: I think the case `#blockhash(.List, 3, 3, 3)` is undefined?

File: data.md [hunk 5]
```diff
@@ -433,16 +433,19 @@ A cons-list is used for the EVM wordstack.
...
  // --------------------------------------------------------------
     rule (W0 : WS)  [ N := W ] => W  : WS                             requires N ==Int 0
-    rule .WordStack [ N := W ] => 0  : (.WordStack [ N -Int 1 := W ]) requires N >Int 0
     rule (W0 : WS)  [ N := W ] => W0 : (WS [ N -Int 1 := W ])         requires N >Int 0
 ```
 
+```{.k .symbolic}
+    rule #Ceil(WS[N]) => {((0 <=Int N) andBool (N <Int #sizeWordStack(WS))) #Equals true}  [anywhere]
+    rule #Ceil(WS[ N := W ]) => {((0 <=Int N) andBool (N <Int #sizeWordStack(WS))) #Equals true}  [anywhere]
+```
```
Comment (ehildenb) [inline data.md]: Format these nicely (line up the `=>`, and the `andBool`, and the `{ ... #Equals ... }`).

Also, add an explanation above these lines that this is defining the undefinedness of these operators.

File: evm.md (hunk 4 above)
Comment (yzhang90) [reply evm.md]: I removed the attribute for now. Let's wait for @dwightguth's response.

File: data.md (hunk 5 above)
Comment (yzhang90) [reply data.md]: done

**Pull request merged**